### ISSUE TITLE
Custom courses

### ISSUE THREAD

**Issue by alexmojaki** [alexmojaki/futurecoder#29]
@alexmojaki I haven't been able to check the demo, will try later. I am using python for business analytics which means I pretty much have to train each person that comes into my team or that wants to use my tools in my company, so I train a lot of people on python. One reason I have to check futurecoder for me is to see if I can create custom courses. Me and my team have developed a python library for data analytics and I would love to have a standard course for new team mates for instance.

To give you a rough idea of this industry. Enterprise "UI" based tools to do what me and my team do with Python are between $4,000 (Alteryx) and $9,000 (DataIku) per user per year.

_Originally posted by @acivitillo in https://github.com/alexmojaki/futurecoder/issues/23#issuecomment-667637269_

**Comment by alexmojaki** [comment]
@acivitillo can you tell me a bit more about your use case? Are you training people from scratch who have no experience in Python? Do you want a course that teaches people who already know Python how to specifically use your library? Do you want a course that teaches Python from scratch (i.e. similar to the current course) but the theme/style (the exercises, the examples, etc.) is more business and maths oriented?

Currently there is no real support for custom courses. What you can do is fork and clone the repository, write your own content in the fork, and then host the server yourself (possibly just locally on each student's machine). How does that sound?

Writing content with your own exercises and stuff is not too difficult. You can take a look at the current course content to get an idea of what it's like: https://github.com/alexmojaki/futurecoder/tree/master/backend/main/chapters . I will write some documentation on the API and how it works. Running the server is easy.

**Comment by acivitillo** [comment]
Yes I am training business analysts from scratch. Those are people that normally would use SQL, MS Access, VBA, Alteryx and I am taking the bet of training them on Python instead. It's working OK so far but we lack a formalized introduction "certification" process, so this could be it. We definitely need custom courses because some material will be internal to our company (i.e. tied to our infrastructure). I'll have a chat with my team next week and see if we want to proceed and check this out. I like your approach to writing the course using python classes, we already do our documentation in markdown on git so it will be easier to write courses as *.py files.

**Comment by aninuth** [comment]
I'd love to help out with designing courses focused on math/data analytics! I would be happy to help @acivitillo with his idea, and would also be interested in further contributing to FutureCoder with this focus.

**Comment by alexmojaki** [comment]
@acivitillo I've documented the API for writing course content here: https://github.com/alexmojaki/futurecoder/blob/master/how_to_contribute.md#how-to-implement-course-content-in-python
